src/Stratis.Bitcoin/Consensus/ConsensusManger.cs

```diff
@@ -227,123 +260,128 @@ private void ProcessDownloadedBlock(ChainedHeaderBlock chainedHeaderBlock)
+                    foreach (int peerId in peersToBan)
+                    {
+                        if (this.peersByPeerId.TryGetValue(peerId, out INetworkPeer peer))
+                            this.peerBanning.BanAndDisconnectPeer(peer.RemoteSocketEndpoint, validationResult.BanDurationSeconds, validationResult.BanReason);
+                    }
                 }
             }
 
             this.logger.LogTrace("(-)");
         }
 
+        /// <summary>
+        /// Handles a situation when partial validation of a block was successful. Informs CHT about
+        /// finishing partial validation process and starting a new partial validation or full validation.
+        /// </summary>
+        /// <param name="chainedHeaderBlock">Block which validation was successful.</param>
         private async Task OnPartialValidationSucceededAsync(ChainedHeaderBlock chainedHeaderBlock)
         {
             this.logger.LogTrace("({0}:'{1}')", nameof(chainedHeaderBlock), chainedHeaderBlock);
 
-            List<ChainedHeader> nextHeadersToValidate;
+            List<ChainedHeaderBlock> chainedHeaderBlocksToValidate;
             ConnectBlocksResult connectBlocksResult = null;
 
             using (await this.reorgLock.LockAsync().ConfigureAwait(false))
             {
-                bool reorgRequired;
+                bool fullValidationRequired;
 
                 lock (this.peerLock)
                 {
-                    nextHeadersToValidate = this.chainedHeaderTree.PartialValidationSucceeded(chainedHeaderBlock.ChainedHeader, out reorgRequired);
+                    chainedHeaderBlocksToValidate = this.chainedHeaderTree.PartialValidationSucceeded(chainedHeaderBlock.ChainedHeader, out fullValidationRequired);
                 }
 
-                if (reorgRequired)
+                if (fullValidationRequired)
                 {
-                    connectBlocksResult = await this.FullyValidateAndReorgLockedAsync(chainedHeaderBlock).ConfigureAwait(false);
+                    connectBlocksResult = await this.FullyValidateLockedAsync(chainedHeaderBlock).ConfigureAwait(false);
                 }
             }
 
             if (connectBlocksResult != null)
             {
                 if (connectBlocksResult.PeersToBan != null)
                 {
-                    foreach (int peerId in connectBlocksResult.PeersToBan)
-                    {
-                        var peer = this.connectionManager.FindNodeById(peerId);
-                        ConnectionManagerSettings connectionSettings = this.connectionManager.ConnectionSettings;
-                        this.peerBanning.BanPeer(peer.PeerEndPoint, connectionSettings.BanTimeSeconds);
-                    }
-
                     lock (this.peerLock)
                     {
+                        foreach (int peerId in connectBlocksResult.PeersToBan)
+                        {
+                            if (this.peersByPeerId.TryGetValue(peerId, out INetworkPeer peer))
```

Comment:
I think we don't necessarily need `INetworkPeer` in the dictionary this will be looked up in `peerBanning.BanAndDisconnectPeer`